test/test_train_mp_imagenet.py

```diff
@@ -132,13 +151,30 @@ def train_imagenet():
 
   torch.manual_seed(42)
 
+  devices = (
+      xm.get_xla_supported_devices(
+          max_devices=FLAGS.num_cores) if FLAGS.num_cores != 0 else [])
   device = xm.xla_device()
   model = get_model_property('model_fn')().to(device)
+  writer = SummaryWriter(log_dir=FLAGS.logdir) if FLAGS.logdir else None
   optimizer = optim.SGD(
       model.parameters(),
       lr=FLAGS.lr,
       momentum=FLAGS.momentum,
       weight_decay=5e-4)
+  total_num_devices = len(
+      xm.xla_replication_devices(devices)) if len(devices) > 1 else 1
+  num_training_steps_per_epoch = train_dataset_len // (
+      FLAGS.batch_size * total_num_devices)
+  lr_scheduler = schedulers.wrap_optimizer_with_scheduler(
+      optimizer,
+      scheduler_type=getattr(FLAGS, 'lr_scheduler_type', None),
+      scheduler_divisor=getattr(FLAGS, 'lr_scheduler_divisor', None),
+      scheduler_divide_every_n_epochs=getattr(
+          FLAGS, 'lr_scheduler_divide_every_n_epochs', None),
+      num_steps_per_epoch=num_training_steps_per_epoch,
+      summary_writer=writer if test_utils.is_first_device(
```

Comment:
Here there is only one device.
So maybe:
```Python
test_utils.is_first_device(device, [device])

Reply:
done

Reply:
On second thought: changed is_first_device to have a multiprocess flag to make it more obvious what is going on

Reply:
On third thought: changed back to the original :)
I agree with Davide's point